main.go

```diff
@@ -57,8 +97,45 @@ func main() {
 
 	err = discord.Open()
 	utils.Assert("Error opening connection to Discord", err)
+
+	for i := 0; i < len(discord.State.Guilds); i++ {
+		currentguild := discord.State.Guilds[i]
+		println(currentguild.ID)
+		channellist, err := discord.GuildChannels(currentguild.ID)
+		utils.Assert("error retrieving channellist", err)
+		println(channellist)
+		var (
+			messagechannel string
+		)
+		for a := 0; a < len(channellist); a++ {
+			if channellist[a].Type == discordgo.ChannelTypeGuildText {
+				messagechannel = channellist[a].ID
+			}
+			if channellist[a].Name == "general" {
+				message, err := discord.ChannelMessageSend(channellist[a].ID, "Hello Fish!")
+				utils.Assert("error sending message", err)
+				println(message.ID)
+			}
+		}
+		for r := 0; r < len(currentguild.Roles); r++ {
+			currentrole := currentguild.Roles[r]
+			println(currentrole.Name)
+			println(currentrole.ID)
+			if currentrole.Name == "admin" {
+				println(messagechannel)
```

Comment:
At this point in execution, 'messagechannel' is merely the last channelID in the list of channels from the server.  Is that the goal here?